src/main/java/mekanism/common/item/gear/ItemMekaTool.java

```diff
@@ -235,19 +239,25 @@ public boolean onBlockStartBreak(ItemStack stack, BlockPos pos, Player player) {
             boolean silk = isModuleEnabled(stack, MekanismModules.SILK_TOUCH_UNIT);
             FloatingLong energyRequired = getDestroyEnergy(stack, state.getDestroySpeed(world, pos), silk);
             if (energyContainer.extract(energyRequired, Action.SIMULATE, AutomationType.MANUAL).greaterOrEqual(energyRequired)) {
+                Set<BlockPos> blocks = new HashSet<>();
+                blocks.add(pos.immutable());
+
+                //Setup initial set for blasting
+                IModule<ModuleBlastingUnit> blastingUnit = getModule(stack, MekanismModules.BLASTING_UNIT);
+                if (blastingUnit != null && blastingUnit.isEnabled() && !(state.getBlock() instanceof BlockBounding) && !player.isShiftKeyDown()) {
+                    Direction direction = getPlayerPOVHitResult(world, player, ClipContext.Fluid.NONE).getDirection();
+                    blocks.addAll(ModuleBlastingUnit.findPositions(world, pos, direction, blastingUnit.getCustomInstance().getBlastRadius()));
+                }
+
                 IModule<ModuleVeinMiningUnit> veinMiningUnit = getModule(stack, MekanismModules.VEIN_MINING_UNIT);
                 //Even though we now handle breaking bounding blocks properly, don't allow vein mining them
                 if (veinMiningUnit != null && veinMiningUnit.isEnabled() && !(state.getBlock() instanceof BlockBounding)) {
-                    boolean isOre = state.is(MekanismTags.Blocks.ATOMIC_DISASSEMBLER_ORE);
-                    //If it is extended or should be treated as an ore
-                    if (isOre || veinMiningUnit.getCustomInstance().isExtended()) {
-                        //Don't include bonus energy required by efficiency modules when calculating energy of vein mining targets
-                        FloatingLong baseDestroyEnergy = getDestroyEnergy(silk);
-                        Set<BlockPos> found = ModuleVeinMiningUnit.findPositions(state, pos, world, isOre ? -1 : veinMiningUnit.getCustomInstance().getExcavationRange());
-                        MekanismUtils.veinMineArea(energyContainer, world, pos, (ServerPlayer) player, stack, this, found,
-                              hardness -> getDestroyEnergy(baseDestroyEnergy, hardness), distance -> 0.5 * Math.pow(distance, isOre ? 1.5 : 2), state);
-                    }
+                    blocks.addAll(ModuleVeinMiningUnit.findPositions(world, blocks, veinMiningUnit.getCustomInstance().isExtended(), veinMiningUnit.getCustomInstance().getExcavationRange()));
                 }
+                //Don't include bonus energy required by efficiency modules when calculating energy of vein mining targets
+                FloatingLong baseDestroyEnergy = getDestroyEnergy(silk);
+                MekanismUtils.veinMineArea(energyContainer, world, pos, (ServerPlayer) player, stack, this, blocks,
+                        hardness -> getDestroyEnergy(baseDestroyEnergy, hardness), (distance, isOre) -> 0.5 * Math.pow(distance, isOre ? 1.5 : 2), state);
```

Comment:
This should only be done if `blocks` is not empty, and I don't think the source position needs to be being added to be added to the blocks.

Reply:
I can add that check. I only added the source block first so that it maintained the order, just felt cleaner that way. But it's not even mining it with the vein mine so it doesn't need to be in there except when vein mining with blasting off, so I could just move it to in the vein mining block.

Reply:
Ah good point about the order, but that also means that blocks should be using a `LinkedHashSet` instead of a `HashSet` so that it actually persists the order.

Reply:
If the order really mattered I would agree, but it was really just so it felt cleaner. It doesn't need to be added at all if area mining is on as it will be one of the area mined ones. I'm thinking just add it to the vein mining if block so that the blocks.isEmpty won't fail when both area and vein mining are disabled.

Reply:
If I recall the order does actually matter because if you do not have energy to mine all the blocks it wants to try and mine then it will look and behave "wrong" compared to how the user likely expects as it will basically end up not just breaking the closest blocks.

Reply:
The order matters for vein mining. The order does not really matter for the area mined ones. The target block gets mined separately so that will always get mined regardless of the vein mining. But all of the area mined ones go into the initial ordered blocks for vein mining and while it technically could matter if you were running into the max amount to mine limit, it would only be that the logs at distance 7 got mined before the coal at distance 7(or vice versa or potentially even log then coal then log at distance 7 if you got two different trees in your area mine). It could potentially be cool to have it go in distance from the target block when area mining when you run out of energy but that would require reworking the order they are added from the area method, and I don't think it's worth it.

Reply:
oh, you're right. I was thinking about the set that findPositions was returning for area/vein, blocks should be ordered when it gets passed to the vein mining util. But I think I'm gonna just set blocks instead of addAll since that can get a little big though.